core/src/main/scala/io/finch/ToService.scala

```diff
@@ -111,3 +87,58 @@ object ToService {
     }
   }
 }
+
+trait LowPriorityToService {
+
+  implicit def hlistNegTS[A, EH <: Endpoint[A], ET <: HList, CTH <: Coproduct, CTT <: HList](implicit
```

Comment:
I'm thinking there is a way to unify two cases: when a single content-type or a coproduct content-type is available. Maybe we can define `ToService` in terms of `ContentTypeNegotiation` (or `NegotiateToResponse`) instead of `ToResponse` and also derive an instance of `ContentTypeNegotiation` for a case when single content-type is available. This way we can posily get rid of code duplication. What do you think about this?

Reply:
Yep, I had this in mind, though didn't want to add an overhead penalty for those who use single type.

Reply:
Why it has to be an overhead? I'd imagine `Bootstrap.serve[CT]` (single content-type) should still work.

Reply:
overhead in terms of allocations per request

Reply:
I'd say it's worth it. Also I doubt it's going to be on a per-request basis as we only need this materialize the `NegotiateToReponse` once for a given content-type/output type pair.

Reply:
Yeah, but it at least doubles amount of `ToResponse` instances generated per request

Reply:
My guess is it's just a drop in the ocean. We can always run the `ToServiceBenchmark` to get some concrete numbers.